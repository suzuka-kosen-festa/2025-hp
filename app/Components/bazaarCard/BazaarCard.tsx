/** @jsxImportSource @emotion/react */
import type { ReactNode } from "react";
import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Typography from "@mui/material/Typography";
import * as React from "react";
import { ja } from "../../locales/ja";

export interface BazaarComponentProps {
  image: string
  teamName: string
  bazaarName: string
  description: string
}

function BazaarCard(props: BazaarComponentProps): ReactNode {
  const { image, teamName, bazaarName, description } = props;
  const handleImageError = (e: React.SyntheticEvent<HTMLImageElement, Event>) => {
    e.currentTarget.src = "/images/not-found.png";
  };
  return (
    <Card sx={{ maxWidth: 345 }}>
      <Box sx={{
        height: "350px",
        display: "flex",
        flexDirection: "column",
        alignItems: "stretch",
        justifyContent: "flex-start",
      }}
      >
        <CardMedia
          component="img"
          height="auto"
          image={image}
          alt={ja.bazaarCard.imageAlt(bazaarName)}
          onError={handleImageError}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="div">
            {ja.bazaarCard.name(bazaarName, teamName)}
          </Typography>
          <Typography variant="body2" sx={{ color: "text.secondary" }}>
            {description}
          </Typography>
        </CardContent>
      </Box>
    </Card>
  );
}

export default BazaarCard;
