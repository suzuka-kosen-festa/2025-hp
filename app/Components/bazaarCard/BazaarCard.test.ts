// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";
import * as stories from "./BazaarCard.stories";

const { Default } = composeStories(stories);

it("bazaarCard snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
