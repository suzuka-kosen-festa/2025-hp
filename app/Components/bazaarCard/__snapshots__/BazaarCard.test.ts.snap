// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bazaarCard snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1i3faf7-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiBox-root css-pqk3o7"
    >
      <img
        alt="バザー名のイメージ"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1kuwa16-MuiCardMedia-root"
        height="auto"
        src="/public/images/not-found.png"
      />
      <div
        class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1f9xtph-MuiTypography-root"
        >
          バザー名-チーム名
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
        >
          説明文
        </p>
      </div>
    </div>
  </div>
</div>
`;
